sympy/integrals/prde.py

```diff
@@ -1038,6 +1062,47 @@ def is_deriv_k(fa, fd, DE):
             return (ans, result, const)
 
 
+def base_Q(G):
+    """
+    >>> G = [Poly(x**2*sqrt(2) + x, x, extension=True),
+            Poly(x**2*sqrt(3) + 1, x, extension=True)]
+    >>> base_Q(G)
+    ((Rational function field in  over QQ<sqrt(2) + sqrt(3)> with lex order,
+      [ANP([1/2, 0, -9/2, 0], [1, 0, -10, 0, 1], QQ)/ANP([1], [1, 0, -10, 0, 1], QQ),
+          -ANP([-1/2, 0, 11/2, 0], [1, 0, -10, 0, 1], QQ)/ANP([1], [1, 0, -10, 0, 1], QQ)]),
+     QQ<sqrt(2) + sqrt(3)>)
+
+    """
+    from sympy import sfield, AlgebraicField, QQ
+    # algebracic constants
+    algebraic_consts = set()
+    # transcendental constants that are algebraically
+    # independent
+    alg_ind_trans_consts = set()
+
+    for poly in G:
+        trans_consts = list(set(poly.gens))
+        expr_args = poly.as_expr().args
+        # T = [t1, t2, ..., tn]
+        # obtaining algebraically independent
+        # for ex. if π and π^2 both are in poly.as_expr() then
+        # both should be added
+        T = set([num**(i.as_coeff_exponent(num)[1])
+            for num in trans_consts for i in expr_args])
```

Comment:
These seem to be the monomials as expressions. Would it be possible to use `monoms`?